enterprise/admin/forms.py

```diff
@@ -331,6 +338,19 @@ def _validate_program(self):
                 mode=course_mode, program_title=program.get("title"), modes=", ".join(available_modes)
             ))
 
+    def _validate_reason(self):
+        """
+        Verify that the reason field is populated if the new learner(s) is being enrolled
+        in a course or program.
+        """
+        program = self.cleaned_data.get(self.Fields.PROGRAM)
```

Comment:
NIT: Can you please remove the program field check from here? My current PR will remove the Program UUID field from the form.